src/Worker.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Equip\Queue;
+
+use Equip\Queue\Driver\DriverInterface;
+use Exception;
+
+class Worker
+{
+    /**
+     * @var DriverInterface
+     */
+    private $driver;
+
+    /**
+     * @var Event
+     */
+    private $event;
+
+    /**
+     * @var array
+     */
+    private $handlers;
+
+    /**
+     * @var array
+     */
+    private $options = [];
+
+    /**
+     * @param DriverInterface $driver
+     * @param Event $event
+     * @param array $handlers
+     * @param array $options
+     */
+    public function __construct(
+        DriverInterface $driver,
+        Event $event,
+        array $handlers = [],
+        array $options = []
+    ) {
+        $this->driver = $driver;
+        $this->event = $event;
+        $this->handlers = $handlers;
+        $this->options = array_merge($this->options, $options);
+    }
+
+    /**
+     * Consumes messages off of the queue
+     *
+     * @param string $queue
+     */
+    public function consume($queue)
+    {
+        declare (ticks = 1);
```

Comment:
What's the purpose of using this if we're not [registering a tick handler](http://php.net/manual/en/control-structures.declare.php#control-structures.declare.ticks) ?